validation/src/test/scala/hmda/validation/rules/lar/validity/V430Spec.scala

```diff
@@ -0,0 +1,42 @@
+package hmda.validation.rules.lar.validity
+
+import hmda.model.fi.lar.{ Loan, LoanApplicationRegister }
+import hmda.validation.dsl.{ Failure, Success }
+import hmda.validation.rules.EditCheck
+import hmda.validation.rules.lar.LarEditCheckSpec
+import org.scalacheck.Gen
+
+class V430Spec extends LarEditCheckSpec with BadValueUtils {
+  property("Valid if preapprovals = 3") {
+    forAll(larGen) { lar =>
+      val validLar = lar.copy(preapprovals = 3)
+      validLar.mustPass
+    }
+  }
+
+  val validPurpose: Gen[Int] = intOutsideRange(1, 2)
+
+  property("Valid if loan purpose not 1 or 2") {
+    forAll(larGen, validPurpose) { (lar, x) =>
+      val validLoan = lar.loan.copy(purpose = x)
+      val validLar = lar.copy(loan = validLoan)
+      validLar.mustPass
+    }
+  }
+
+  val invalidPurpose: Gen[Int] = Gen.oneOf(1, 2)
+  val invalidPreApproval: Gen[Int] = Gen.choose(Int.MinValue, Int.MaxValue).filter(_ != 3)
```

Comment:
It looks like this pattern `Gen.choose(Int.MinValue, Int.MaxValue).filter(_ != x)` is used frequently enough that it could be useful as a function in the `BadValueUtils` class. Want to extract it?